### PR TITLE
Feature/61 authentication audit

### PR DESCRIPTION
**Pull request by eddiejaoude**

### CODE REVIEW

**Comment by brettminnie** [comment]
can you rebase this against develop so we can remove the duplicate commits please

**Comment by eddiejaoude** [comment]
Sure will do! :+1:

**Comment by eddiejaoude** [comment]
Done. Merged in `develop`

File: src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php [hunk 1]
```diff
@@ -12,6 +16,28 @@
  */
 class DashboardType extends AbstractType
 {
+
+    /**
+     * @var SecurityContextInterface
+     */
+    private $securityContext;
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php]: Visibility should be protected

File: src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * Class AuthenticationListener
+ * @package DashboardHub\Bundle\AppBundle\Listener
+ */
+class AuthenticationListener
+{
+
+    /**
+     * @var SecurityContext
+     */
+    private $securityContext;
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php]: Protected

File: src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * @param EntityManager   $em
+     * @param SecurityContext $securityContext
+     */
+    public function __construct(EntityManager $em, SecurityContext $securityContext)
+    {
+        $this->securityContext = $securityContext;
+        $this->em = $em;
+    }
+
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php]: Docblock

File: src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php [hunk 4]
```diff
@@ -46,11 +76,25 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'save',
                 'submit',
                 array('label' => 'Save')
+            )
+            ->addEventListener(
+                FormEvents::POST_SET_DATA,
+                array($this, 'onPostSetData')
             );
     }
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php]: Docblock

**Comment by eddiejaoude** [comment]
The tests passed, but code coverage generation failed. Rebuilding it again

**Comment by eddiejaoude** [comment]
PHPSpec:

```
$ ./bin/phpspec run --config=test/phpspec.yml -vvv
                                 100%                                  41
11 specs
41 examples (41 passed)
7803ms
```

Behat:
```
4 scenarios (4 passed)
11 steps (11 passed)
0m2.54s (45.85Mb)
```

Failure in sending Code Coverage:
```
$ php ocular.phar code-coverage:upload --format=php-clover test/build/coverage.xml
#!/usr/bin/env php
Uploading code coverage for repository "g/DashboardHub/PipelineDashboard" and revision "3e391c6d4f8d6c6da174b268e6bc9c869a6ddc11"... 
                                                                               
  [Guzzle\Http\Exception\CurlException]                                        
  [curl] 23: Failed writing body (0 != 16) [url] https://scrutinizer-ci.com/a  
  pi/repositories/g/DashboardHub/PipelineDashboard/data/code-coverage          
                                                                               
code-coverage:upload [--api-url="..."] [--repository="..."] [--revision="..."] [--format="..."] [--parent="..."] coverage-file
The command "php ocular.phar code-coverage:upload --format=php-clover test/build/coverage.xml" exited with 1.
```

I will investigate.

File: src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php (hunk 1 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php]: :+1:

File: src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php (hunk 2 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php]: Doh! I will remember this soon :wink:

File: src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php (hunk 3 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php]: ?

```
/**
  * @param EntityManager   $em
  * @param SecurityContext $securityContext
  */
  public function __construct(EntityManager $em, SecurityContext $securityContext)
```

**Comment by eddiejaoude** [comment]
I get the same locally:

```
➜  PipelineDashboard git:(feature/61-authentication-audit) wget https://scrutinizer-ci.com/ocular.phar
--2015-03-09 10:11:08--  https://scrutinizer-ci.com/ocular.phar
Resolving scrutinizer-ci.com... 46.105.232.227
Connecting to scrutinizer-ci.com|46.105.232.227|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://s3-eu-west-1.amazonaws.com/scrutinizer-static/ocular.phar [following]
--2015-03-09 10:11:08--  https://s3-eu-west-1.amazonaws.com/scrutinizer-static/ocular.phar
Resolving s3-eu-west-1.amazonaws.com... 54.231.136.96
Connecting to s3-eu-west-1.amazonaws.com|54.231.136.96|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3929473 (3.7M) [application/octet-stream]
Saving to: 'ocular.phar'

ocular.phar                                                           100%[=========================================================================================================================================================================>]   3.75M  2.15MB/s   in 1.7s   

2015-03-09 10:11:10 (2.15 MB/s) - 'ocular.phar' saved [3929473/3929473]

➜  PipelineDashboard git:(feature/61-authentication-audit) ✗ php ocular.phar code-coverage:upload --format=php-clover test/build/coverage.xml 
#!/usr/bin/env php
Uploading code coverage for repository "g/DashboardHub/PipelineDashboard" and revision "d24781842d8ee7adc26e07d096985fb8bdd76a03"... 

                                                                                                                                                  
  [Guzzle\Http\Exception\CurlException]                                                                                                           
  [curl] 23: Failed writing body (0 != 16) [url] https://scrutinizer-ci.com/api/repositories/g/DashboardHub/PipelineDashboard/data/code-coverage  
                                                                                                                                                  


code-coverage:upload [--api-url="..."] [--repository="..."] [--revision="..."] [--format="..."] [--parent="..."] coverage-file
```

**Comment by eddiejaoude** [comment]
It appears Scrutinizer is down...

screenshot 2015-03-09 10 14 22

**Comment by brettminnie** [comment]
Joy...

On Mon, Mar 9, 2015 at 10:16 AM, Eddie Abou-Jaoude <notifications@github.com

**Comment by eddiejaoude** [comment]
@brettminnie Scrutinizer is backup & all is green. Please review when you get a sec.

File: src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php (hunk 3 above)
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Listener/AuthenticationListener.php]: missing on public function onSecurityInteractiveLogin(InteractiveLoginEvent $event)

**Comment by brettminnie** [comment]
Merging, one missing docblock, make a note of it
`public function onSecurityInteractiveLogin(InteractiveLoginEvent $event)`

**Comment by eddiejaoude** [comment]
> Merging, one missing docblock, make a note of it
public function onSecurityInteractiveLogin(InteractiveLoginEvent $event)

Thanks for merging & comment - already done in next PR :smile:

**Comment by eddiejaoude** [comment]
> already done in next PR

I should stop using that excuse & get it right first time :wink:

**Comment by brettminnie** [comment]
We all miss those :D

On Tue, Mar 10, 2015 at 9:01 AM, Eddie Abou-Jaoude <notifications@github.com

**Pull request merged**